src/Chapter3.hs

```diff
@@ -480,6 +519,36 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+newtype Castle = MkCastle
+    { castleName :: String
+    }
+
+data Building = Church | Library
+
+newtype House = MkHouse
+    { housePeople :: Int
```

Comment:
Interesting approach! 
However, you will need to make some function (for example `mkHouse`) to avoid building houses with more than 4 people. You need to restrict usage of the Constructor and allow to create `House`s only with your `mkHouse` function.
You can read a bit more on the pattern in here: https://kowainik.github.io/posts/haskell-mini-patterns#smart-constructor

Generally, in here, you could restrict that on type level, by using something else than `Int` 🙂